src/PBR/Application/PBR/PBRApplication.cs

```diff
@@ -0,0 +1,568 @@
+
+                                commandList.SetPipeline(pipeline);
+                                commandList.SetComputeResourceSet(0, irradianceResourceSet);
+                                commandList.Dispatch(kIrradianceMapSize / 32, kIrradianceMapSize / 32, 6);
+
+                                commandList.CopyTexture(temporaryIrradianceEnvironmentMapTexture,
+                                    irradianceEnvironmentMapTexture);
+
+                                commandList.End();
+                                GraphicsDevice.SubmitCommands(commandList);
+                                GraphicsDevice.WaitForIdle();
+                            }
+                        }
+                    }
+
+                    var specularBRDFLUTTexture = factory.CreateTexture(TextureDescription.Texture2D(kBRDF_LUT_Size,
+                        kBRDF_LUT_Size, 1, 1, PixelFormat.R16_G16_Float, TextureUsage.Storage | TextureUsage.Sampled));
+                    {
+                        // compute SpecularBRDF_LUT
+                        var spbrdfShaderSrc = ReadEmbeddedAssetBytes("spbrdf_cs.glsl");
+
+                        var spbrdfUniformLayoutDescription = new ResourceLayoutDescription(
+                            new ResourceLayoutElementDescription("LUT", ResourceKind.TextureReadWrite,
+                                ShaderStages.Compute)
+                        );
+
+                        using (var shader =
+                            factory.CreateFromSpirv(new ShaderDescription(ShaderStages.Compute, spbrdfShaderSrc, "main",
+                                Debug)))
+                        using (var spbrdfUniformLayout = factory.CreateResourceLayout(spbrdfUniformLayoutDescription))
+                        using (var spbrdfResourceSet =
+                            factory.CreateResourceSet(new ResourceSetDescription(spbrdfUniformLayout,
+                                specularBRDFLUTTexture)))
+                        using (var pipeline =
+                            factory.CreateComputePipeline(
+                                new ComputePipelineDescription(shader, spbrdfUniformLayout, 32, 32, 1)))
+                        using (var commandList = factory.CreateCommandList())
+                        {
+                            commandList.Begin();
+
+                            commandList.SetPipeline(pipeline);
+                            commandList.SetComputeResourceSet(0, spbrdfResourceSet);
+                            commandList.Dispatch(kBRDF_LUT_Size / 32, kBRDF_LUT_Size / 32, 6);
+
+                            commandList.End();
+                            GraphicsDevice.SubmitCommands(commandList);
+                            GraphicsDevice.WaitForIdle();
+                        }
+                    }
+
+                    if (GraphicsDevice.GetVulkanInfo(out var info))
+                    {
+                        info.TransitionImageLayout(specularEnvironmentMapTexture,
```

Comment:
I don't think these are supposed to be necessary.  I don't think I've fixed anything yet that would make them redundant but there may be something yet to fix in the library re: transitioning states after compute shaders are run?